src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -0,0 +1,1378 @@
+        }
+        private static IReturnStatement CreateBoundReturnStatementOperation(BoundReturnStatement boundReturnStatement)
+        {
+            Lazy<IOperation> returnedValue = new Lazy<IOperation>(() => (IOperation)Create(boundReturnStatement.ExpressionOpt));
+            bool isInvalid = boundReturnStatement.HasErrors;
+            SyntaxNode syntax = boundReturnStatement.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            return new LazyReturnStatement(returnedValue, isInvalid, syntax, type, constantValue);
+        }
+        private static IReturnStatement CreateBoundYieldReturnStatementOperation(BoundYieldReturnStatement boundYieldReturnStatement)
+        {
+            Lazy<IOperation> returnedValue = new Lazy<IOperation>(() => (IOperation)Create(boundYieldReturnStatement.Expression));
+            bool isInvalid = boundYieldReturnStatement.HasErrors;
+            SyntaxNode syntax = boundYieldReturnStatement.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            return new LazyReturnStatement(returnedValue, isInvalid, syntax, type, constantValue);
+        }
+        private static ILockStatement CreateBoundLockStatementOperation(BoundLockStatement boundLockStatement)
+        {
+            Lazy<IOperation> lockedObject = new Lazy<IOperation>(() => (IOperation)Create(boundLockStatement.Argument));
+            Lazy<IOperation> body = new Lazy<IOperation>(() => (IOperation)Create(boundLockStatement.Body));
+            bool isInvalid = boundLockStatement.HasErrors;
+            SyntaxNode syntax = boundLockStatement.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            return new LazyLockStatement(lockedObject, body, isInvalid, syntax, type, constantValue);
+        }
+        private static IInvalidStatement CreateBoundBadStatementOperation(BoundBadStatement boundBadStatement)
+        {
+            Lazy<ImmutableArray<IOperation>> children = new Lazy<ImmutableArray<IOperation>>(() => boundBadStatement.ChildBoundNodes.SelectAsArray(n => Create(n)));
+            bool isInvalid = boundBadStatement.HasErrors;
+            SyntaxNode syntax = boundBadStatement.Syntax;
+            ITypeSymbol type = null;
+            Optional<object> constantValue = default(Optional<object>);
+            return new LazyInvalidStatement(children, isInvalid, syntax, type, constantValue);
+        }
+        private static IVariableDeclarationStatement CreateBoundLocalDeclarationOperation(BoundLocalDeclaration boundLocalDeclaration)
+        {
+            Lazy<ImmutableArray<IVariableDeclaration>> declarations = new Lazy<ImmutableArray<IVariableDeclaration>>(() => GetVariableDeclarationStatementVariables(boundLocalDeclaration));
+            bool isInvalid = boundLocalDe
```

Comment:
@genlu @mavasani @333fred do you guys have pending work on IOperation? this includes the Evaluation order and IVariableDecl changes.